### PR TITLE
Implement GPUComputePassEncoder functions

### PR DESCRIPTION
**Pull request by imiklos**
Implement the `dispatch`, `endPass`, `setBindGroup`, `setPipeline` functions of `GPUComputePassEncoder`.

---

- [x] `./mach build -d` does not report any errors
- [x] `./mach test-tidy` does not report any errors
- [x] These changes addresses a part of #24706

cc @kvark @jdm @zakorgy

### CODE REVIEW

**Comment by imiklos** [comment]
I've removed the `mach` and `nix` crates from the ignore duplicates list.

**Review by kvark — APPROVED** [review]

File: Cargo.lock [hunk 1]
```diff
@@ -6758,10 +6709,9 @@ dependencies = [
 [[package]]
 name = "wgpu-core"
 version = "0.1.0"
-source = "git+https://github.com/gfx-rs/wgpu#d4a46cb60e71d6b108e9052cc581e52e44a16a44"
+source = "git+https://github.com/imiklos/wgpu?branch=command_encoder_public#079da18c533011722ad836c01b58511d3ee870df"
```
[the remark this answers is not in the document]
Comment (imiklos) [reply Cargo.lock]: Only temporary, just for the review. I will update this. This patch is depends on this pull request: https://github.com/gfx-rs/wgpu/pull/496

**Review by jdm — APPROVED** [review]

**Comment by imiklos** [comment]
I've changed back the `wgpu-core` from the forked one and used the already exposed `ffi` functions for encode passes as @kvark mentioned.

**Comment by jdm** [comment]
@bors-servo retry
* #24726

**Comment by jdm** [comment]
@bors-servo retry
* #24726

**Pull request merged**